gcloud/monitoring/test_client.py

```diff
@@ -157,6 +157,127 @@ def test_metric_descriptor_factory(self):
         self.assertEqual(descriptor.description, DESCRIPTION)
         self.assertEqual(descriptor.display_name, '')
 
+    def test_metric_factory(self):
+        TYPE = 'custom.googleapis.com/my_metric'
+        LABELS = {
+            'instance_name': 'my-instance'
+        }
+
+        client = self._makeOne(project=PROJECT, credentials=_Credentials())
+        client.connection = _Connection()   # For safety's sake.
+        metric = client.metric(TYPE, LABELS)
+        self.assertEqual(metric.type, TYPE)
+        self.assertEqual(metric.labels, LABELS)
+
+    def test_resource_factory(self):
+        TYPE = 'https://cloud.google.com/monitoring/api/resources'
+        LABELS = {
+            'instance_id': 'my-instance-id',
+            'zone': 'us-central1-f'
+        }
+
+        client = self._makeOne(project=PROJECT, credentials=_Credentials())
+        client.connection = _Connection()   # For safety's sake.
+        resource = client.resource(TYPE, LABELS)
+        self.assertEqual(resource.type, TYPE)
+        self.assertEqual(resource.labels, LABELS)
+
+    def test_timeseries_factory(self):
+        import datetime
+        from gcloud._testing import _Monkey
+        import gcloud.monitoring.client
+        METRIC_TYPE = 'custom.googleapis.com/my_metric'
+        MY_CUMULATIVE_METRIC = 'custom.googleapis.com/my_cumlative_metric'
+        METRIC_LABELS = {
+            'instance_name': 'my-instance'
+        }
+
+        RESOURCE_TYPE = 'https://cloud.google.com/monitoring/api/resources'
+        RESOURCE_LABELS = {
+            'project_id': 'my-project-id',
+            'instance_id': 'my-instance-id',
+            'zone': 'us-central1-f'
+        }
+
+        VALUE = 42
+        VALUE2 = 43
+        VALUE3 = 44
+
+        TIME1 = datetime.datetime.utcnow()
+        TIME2 = datetime.datetime.utcnow()
+        TIME3 = datetime.datetime.utcnow()
+        RESET_TIME = datetime.datetime.utcnow()
+        TIME4 = datetime.datetime.utcnow()
+
+        # Construct a time series assuming a gauge metric without a start time
+        client = self._makeOne(project=PROJECT, credentials=_Credentials())
+        client.connection = _Connection()   # For safety's sake.
+        metric = client.metric(METRIC_TYPE, METRIC_LABELS)
+        resource = client.resource(RESOURCE_TYPE, RESOURCE_LABELS)
+        timeseries = client.time_series(metric, resource, VALUE,
+                                        end_time=TIME1)
+
+        self.assertEqual(timeseries.metric, metric)
+        self.assertEqual(timeseries.resource, resource)
+        self.assertEqual(len(timeseries.points), 1)
+        self.assertEqual(timeseries.points[0].value, VALUE)
+        self.assertIsNone(timeseries.points[0].start_time)
+        self.assertEqual(timeseries.points[0].end_time, TIME1)
+
+        # Construct a time series assuming a guage metric with an explicit
```

Comment:
guage → gauge

A gauge will require that the start time and end time are identical.